x-pack/plugins/elastic_assistant/server/ai_assistant_service/index.ts

```diff
@@ -287,8 +290,9 @@ export class AIAssistantService {
       opts.spaceId
     );
 
+    const licensing = await opts.licensing;
     // If space level resources initialization failed, retry
-    if (!initialized && error) {
+    if (!initialized && error && hasAIAssistantLicense(licensing.license)) {
```

Comment:
I'm wondering if we could move this check to be the first line of this function. Why run the `getSpaceResourcesInitializationPromise` if no license?
```
if( !hasAIAssistantLicense(licensing.license)) return null;
```

Reply:
Doing this I think may also prevent this "warning":
```
[WARN ][plugins.elasticAssistant.service] Error initializing AI assistant resources: Platinum, Enterprise or trial license needed
```